pkgs/by-name/va/vatprism/package.nix

```
@@ -0,0 +1,98 @@
+{
+  lib,
+  stdenv,
+  jdk,
+  maven,
+  makeWrapper,
+  fetchFromGitHub,
+  libGL,
+  libxkbcommon,
+  wayland,
+  fontconfig,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  libXxf86vm,
+  libXtst,
+  copyDesktopItems,
+  makeDesktopItem,
+}:
+let
+  libPath = lib.makeLibraryPath [
+    libGL
+    libxkbcommon
+    wayland
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    libXxf86vm
+    libXtst
+    fontconfig
+  ];
+in
+maven.buildMavenPackage rec {
+  pname = "vatprism";
+  version = "0.3.5";
+  src = fetchFromGitHub {
+    owner = "marvk";
+    repo = "vatprism";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ofEwHUCm79roHe2bawmKFw2QHhIonnlkFG5nhE6uN+g=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+  # https://github.com/marvk/vatprism/pull/141
+  patches = [ ./0001-Fix-build-on-JDK-21.patch ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "vatprism";
+      desktopName = "VATprism";
+      exec = "vatprism";
+      terminal = false;
+      icon = "vatprism";
+    })
+  ];
+
+  mvnHash = "sha256-9uyNCUqnMgpiwm2kz544pWNB/SkRpASm2Dln0e4yZos=";
```

It fails to build on `aarch64-linux`, looks like the hash changed for aarch64-linux.

[Log from ofborg](https://github.com/NixOS/nixpkgs/pull/329794/checks?check_run_id=27993435718(
```
[INFO] Total time:  34.146 s
[INFO] Finished at: 2024-07-27T07:43:17Z
[INFO] ------------------------------------------------------------------------
buildPhase completed in 36 seconds
Running phase: installPhase
calling 'postInstall' function hook 'copyDesktopItems'
error: hash mismatch in fixed-output derivation '/nix/store/9nq4y95jbqy3ymk8is9g1xwc05vqyy68-vatprism-0.3.5-maven-deps.drv':
         specified: sha256-9uyNCUqnMgpiwm2kz544pWNB/SkRpASm2Dln0e4yZos=
            got:    sha256-x0nFt2C7dZqMdllI1+Io9SPBY2J/dVgBTVb9T24vFFI=
error: 1 dependencies of derivation '/nix/store/npxc84b6bs4makfijnxxz5sj3v1h9kqy-vatprism-0.3.5.drv' failed to build
```

Maybe something like this should work?
```suggestion
  mvnHash = if (stdenv.isLinux && stdenv.isAarch64)
    then "sha256-x0nFt2C7dZqMdllI1+Io9SPBY2J/dVgBTVb9T24vFFI="
    else "sha256-9uyNCUqnMgpiwm2kz544pWNB/SkRpASm2Dln0e4yZos=";
```

---

I'm honestly not setup to test the build/program on aarch64-linux, but I can add this.

---

Honestly a simple `binfmt` settings works wonders.

https://github.com/JohnRTitor/nix-conf/blob/d7ddbdc3753369d8ec69587ae2de4f5d04ded899/pkgs/settings.nix#L3

Though It will take a lot of time to compile if it is not cached and large enough.